src/autowiring/AutoPacket.cpp

```diff
@@ -274,6 +271,48 @@ void AutoPacket::UpdateSatisfactionUnsafe(std::unique_lock<std::mutex> lk, const
 
   // Recursively mark unsatisfiable any single-output arguments on these subscribers:
   std::vector<const AutoFilterArgument*> unsatOutputArgs;
+  auto MarkOutputsUnsat = [&unsatOutputArgs] (const SatCounter& satCounter) {
+    const auto* args = satCounter.GetAutoFilterArguments();
+    for (size_t i = satCounter.GetArity(); i--;) {
+      // Only consider output arguments:
+      if (args[i].is_output)
+        // This output is transitively unsatisfiable, include it for later removal
+        unsatOutputArgs.push_back(&args[i]);
+    }
+  };
+
+  for (auto modifier : disposition.m_modifiers) {
+    if (!modifier.satCounter)
+      continue;
+    auto& satCounter = *modifier.satCounter;
+    if (modifier.is_shared) {
+      if (disposition.m_decorations.size() > 1)
+        throw autowiring_error("An AutoFilter was detected which has single-decorate rvalue shared pointer argument in a graph with multi-decorate outputs");
```

Comment:
On the second thought, the shared pointer modifier can only change the number of decorations from 0 to1 or from 1 to 0. So we should be able to detect this case a little sooner, like above the loop.